main.js

```diff
@@ -0,0 +1,203 @@
+    );
+    invalidMoves.textContent = `There is no tower ${toTower}, try again.  Choose tower 1, 2, or 3.`;
+    validMoves.textContent = "";
+    return false;
+  }
+
+  const fromTowerLength = fromTowerArray.length;
+  const toTowerLength = toTowerArray.length;
+
+  // get disk to move
+  const diskToMove = fromTowerArray[fromTowerLength - 1];
+
+  // get top disc on target tower
+  const topDiscOnTarget = toTowerArray[toTowerLength - 1];
+
+  // check if fromTower is empty
+  if (fromTowerArray.length === 0) {
+    console.log(
+      `There are no disks in tower ${fromTower}, try again.  Choose a tower with disks.`
+    );
+    invalidMoves.textContent = `There are no disks in tower ${fromTower}, try again.  Choose a tower with disks.`;
+    validMoves.textContent = "";
+    return false;
+  } // check if disk to move is larger than top disc on target tower
+  else if (diskToMove > topDiscOnTarget) {
+    console.log(
+      `The disk to move is larger than the top disc on the target tower. Try again.`
+    );
+    invalidMoves.textContent = `Disk ${diskToMove} is larger than disk ${topDiscOnTarget}. Larger disks cannot be placed on smaller disks, try again.`;
+    validMoves.textContent = "";
+    return false;
+  } else {
+    console.log(
+      `Valid move! You can move disk ${diskToMove} from tower ${fromTower} to tower ${toTower}`
+    );
+    validMoves.textContent = `Valid move! You can move disk ${diskToMove} from tower ${fromTower} to tower ${toTower}`;
+    invalidMoves.textContent = "";
+
+    // Get the tower DOM elements using correct selectors
+    const sourceTower = document.querySelector(`.tower${fromTower}`);
+    const targetTower = document.querySelector(`#tower-${toTower}`);
+
+    // Find the specific disk element to move based on the diskToMove value
+    const diskElement = document.querySelector(`.disk-${diskToMove}`);
+
+    // Add disk to the top of the target tower using prepend
+    if (diskElement && targetTower) {
+      targetTower.prepend(diskElement);
+    }
+
+    // Update game state arrays
+    const popDisk = fromTowerArray.pop();
+    const addDisk = toTowerArray.push(popDisk);
+    console.log(`The board now looks like this:`);
+    viewBoard(gameInstance);
+    return true;
+  }
+};
+
+// Listen for Reset/New Game
+const resetGame = reset.addEventListener("click", function (event) {
+  console.log(`Board before reset:`, gameInstance);
+  gameInstance = [...startingBoard];
+  console.log(`The game has been reset, the board is now:`, gameInstance);
+  location.reload();
+});
+
+// check if the game is won
+const checkWinner = function (tower2, tower3) {
+  console.log(`Checking for a winner...`);
+  const winningTower = JSON.stringify([5, 4, 3, 2, 1]);
+  // console.log(`Winning tower JSON:`, winningTower);
+  const tower2Check = JSON.stringify(tower2);
+  // console.log(`Tower 2 JSON:`, tower2Check);
+  const tower3Check = JSON.stringify(tower3);

```

Comment:
you could have use tower.every to compare items, is more performant and readable